README.md

```diff
@@ -61,62 +65,61 @@ rest of the framework. This may be the most appealing design goal, compared to o
+It should always be clear what the Http library is doing from looking at calls in code, without
+worrying about how it has been configured somewhere else. Http should do as few things as possible,
 and should not try to be too smart. Features like connection-retries should be made easy for
 developers to manage, but should not be an automatic, opt-out feature of the library.
 
 ### Don’t Reinvent Wheels
 
 Modern browsers have improved security features, increasingly improving caching ability, and baked-
-in http caching (presuming servers send the correct headers). HTTP should defer as much work as
+in http caching (presuming servers send the correct headers). Http should defer as much work as
 possible to browsers, and should provide logical hooks for developers to manage caching and headers
 themselves.
 
 Composable, Flat API Surface
 
-HTTP may be accompanied by other libraries that provide features like
+Http may be accompanied by other libraries that provide features like
 caching, data serialization, transformation, and validation. These tools should be used by
-developers in conjunction with the HTTP library, and the HTTP library should not know or care how
-they’re being used. In other words, HTTP can’t be configured to automatically leverage these tools
+developers in conjunction with the Http library, and the Http library should not know or care how
+they’re being used. In other words, Http can’t be configured to automatically leverage these tools
 as part of a request/response lifecycle, though developers may define their own wrapper functions to
 make certain processes seamless. Consequently, it’s up to developers to first fetch from a
-proprietary cache before requesting from HTTP, if not relying on implicit browser cache mechanisms.
+proprietary cache before requesting from Http, if not relying on implicit browser cache mechanisms.
 
 
 
 ## Proposed Design
 
-The primary building block of the HTTP implementation is a `Connection`. A `Connection` instance
-represents a single request and response, implements the `Observable` interface. Connections may be
-created by constructing with the `new` keyword, or calling the static `create` method of a
-`Connection` class.
+The primary building block of the Http implementation is a `Connection` object, which is created by
+calling the `http` factory function, or another factory of the application's choosing. A
+`Connection` instance represents a single request and response, and implements the `Observable`
+interface.
 
 Here is a simple example of loading a repository from Github's API:
 
 ```javascript
-import {XHRConnection} from 'http/XHR';
+import {Http} from 'http';
 
 class MyComponent {
-  @Inject(XHRConnection);
-  constructor(XHR:XHRConnection) {
-    XHR
-      .create({url: 'https://api.github.com/repos/angular/angular.js'})
+  @Inject(Http);
+  constructor(http:Http) {
+    http('https://api.github.com/repos/angular/angular.js')
       .subscribe(function(res) {
```

Comment:
I wonder if it would make sense to provide "error()" / "success()" sugar that are popular in 1.x? Might make migration easier. On the other hand, really don't want to have a lot of ways to do the same thing